components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/LogsHandler.java

```diff
@@ -220,31 +220,15 @@ private long getResponseTime(org.apache.synapse.MessageContext messageContext) {
         return responseTime;
     }
 
-    private long buildRequestMessage(org.apache.synapse.MessageContext messageContext) {
+    private long getContentLength(org.apache.synapse.MessageContext messageContext) {
         long requestSize = 0;
         org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext)
                 .getAxis2MessageContext();
         Map headers = (Map) axis2MC.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
         String contentLength = (String) headers.get(HttpHeaders.CONTENT_LENGTH);
         if (contentLength != null) {
             requestSize = Integer.parseInt(contentLength);
-        } else {
```

Comment:
shall we set the content-length to -1 for the scenario where we do not calculate the size. this way we could differentiate it with 0 size one as well.